### PR TITLE
feat(nn4k): support huggingface decode only model local inference 

### PR DESCRIPTION
**Pull request by chenbin11200**
add feature that support local inference with huggingface decode only model

### CODE REVIEW

**Review by didicout — APPROVED** [review]

File: python/nn4k/nn4k/executor/huggingface/base/hf_args.py [hunk 1]
```diff
@@ -105,3 +113,45 @@ def __post_init__(self):
...
                 f"a eval_dataset_path is set but do_eval flag is not set, automatically set do_eval to True"
             )
+
+
+@dataclass
+class HFInferArgs(NNInferenceArgs):
+    delete_heading_new_lines: bool = field(
+        default=False,
+        metadata={
+            "help": "sometimes the output will have a additional question mark and new line marks in the beginning"
```
Comment (leywar) [inline python/nn4k/nn4k/executor/huggingface/base/hf_args.py:123]: an additional  ... ... at the beginning?

File: python/nn4k/nn4k/executor/huggingface/base/hf_args.py [hunk 2]
```diff
@@ -105,3 +113,45 @@ def __post_init__(self):
...
+            "try to get rid of these marks by setting this to True. Different model will have different "
```
Comment (leywar) [inline python/nn4k/nn4k/executor/huggingface/base/hf_args.py:124]: by setting this parameter to True?

File: python/nn4k/nn4k/executor/huggingface/base/hf_llm_executor.py [hunk 3]
```diff
@@ -188,57 +189,87 @@ def load_model(self, args: dict = None, mode=None, **kwargs):
...
         from nn4k.executor.huggingface import HFModelArgs
 
         parser = HfArgumentParser(HFModelArgs)
+        hf_model_args: HFModelArgs
         hf_model_args, *_ = parser.parse_dict(args, allow_extra_keys=True)
 
         self.model_mode = mode
         self._tokenizer = self._hf_tokenizer_loader(hf_model_args)
+
+        # TODO thinking about resume checkpoint in loading steps
```
Comment (leywar) [inline python/nn4k/nn4k/executor/huggingface/base/hf_llm_executor.py:204]: add a `TODO` issue?

**Review by leywar — APPROVED** [review]

File: python/nn4k/nn4k/executor/huggingface/base/hf_llm_executor.py (hunk 3 above)
Comment (chenbin11200) [reply python/nn4k/nn4k/executor/huggingface/base/hf_llm_executor.py:204]: This is done and I forgot to delete that TODO, thanks for reminding.

**Pull request merged**